tracer/src/Datadog.Trace/ClrProfiler/AutoInstrumentation/AspNet/IHttpContext.cs

```diff
@@ -0,0 +1,23 @@
+// <copyright file="IHttpContext.cs" company="Datadog">
+// Unless explicitly stated otherwise all files in this repository are licensed under the Apache 2 License.
+// This product includes software developed at Datadog (https://www.datadoghq.com/). Copyright 2017 Datadog, Inc.
+// </copyright>
+
+using System.Collections;
+using System.ComponentModel;
+
+namespace Datadog.Trace.ClrProfiler.AutoInstrumentation.AspNet
+{
+    /// <summary>
+    /// System.Web.HttpContext interface for ducktyping
+    /// </summary>
+    [Browsable(false)]
+    [EditorBrowsable(EditorBrowsableState.Never)]
+    public interface IHttpContext
+    {
+        /// <summary>
+        /// Gets the items dictionary
+        /// </summary>
+        IDictionary Items { get; }
```

Comment:
Non blocking general question: seeing as this only uses a property (and isn't used in a constraint), should we use a struct with `[DuckCopy]` instead?

Reply:
This is used by the `IThreadContext` so I think it's already optimized for use with the CallTarget machinery. @tonyredondo what do you think?

Reply:
In this case if `IHttpContext` is used only as a property of other ducktype definition and **not as a generic constraint** we can change it to a `[DuckCopy]` struct because there's no method call in the definition.

The difference here will be the removal of an allocation from the autogenerated proxy struct to `IHttpContext` returned by the `IThreadContext.HttpContext` property getter.

Reply:
Thanks for the suggestion Andrew, added in bb9b145b24bb97f6c53fa47ba13b4a8d6e4c000d